pyomo/contrib/doe/examples/fim_doe_tutorial.ipynb

```diff
@@ -57,77 +137,341 @@
+    "            if t in m.t_con:\n",
+    "                return controls[t]\n",
+    "            else:\n",
+    "                # Count how many control points are before the current t;\n",
+    "                # Locate the nearest neighbouring control point before this t\n",
+    "                neighbour_t = max(tc for tc in control_time if tc < t)\n",
+    "                return controls[neighbour_t]\n",
+    "\n",
+    "        mod.T = pyo.Var(\n",
+    "            mod.t, initialize=T_initial, bounds=(300, 700), within=pyo.NonNegativeReals\n",
+    "        )\n",
+    "\n",
+    "        mod.R = 8.31446261815324  # J / K / mole\n",
+    "\n",
+    "        # Define parameters as Param\n",
+    "        mod.A1 = pyo.Var(initialize=theta[\"A1\"])\n",
+    "        mod.A2 = pyo.Var(initialize=theta[\"A2\"])\n",
+    "        mod.E1 = pyo.Var(initialize=theta[\"E1\"])\n",
+    "        mod.E2 = pyo.Var(initialize=theta[\"E2\"])\n",
+    "\n",
+    "        # Concentration variables under perturbation\n",
+    "        mod.C_set = pyo.Set(initialize=[\"CA\", \"CB\", \"CC\"])\n",
+    "        mod.C = pyo.Var(\n",
+    "            mod.C_set, mod.t, initialize=C_init, within=pyo.NonNegativeReals\n",
+    "        )\n",
+    "\n",
+    "        # Time derivative of C\n",
+    "        mod.dCdt = DerivativeVar(mod.C, wrt=mod.t)\n",
+    "\n",
+    "        # Kinetic parameters\n",
+    "        def kp1_init(m, t):\n",
+    "            return m.A1 * pyo.exp(-m.E1 * 1000 / (m.R * m.T[t]))\n",
+    "\n",
+    "        def kp2_init(m, t):\n",
+    "            return m.A2 * pyo.exp(-m.E2 * 1000 / (m.R * m.T[t]))\n",
+    "\n",
+    "        mod.kp1 = pyo.Var(mod.t, initialize=kp1_init)\n",
+    "        mod.kp2 = pyo.Var(mod.t, initialize=kp2_init)\n",
+    "\n",
+    "        def T_control(m, t):\n",
+    "            \"\"\"\n",
+    "            T at interval timepoint equal to the T of the control time point at the beginning of this interval\n",
+    "            Count how many control points are before the current t;\n",
+    "            locate the nearest neighbouring control point before this t\n",
+    "            \"\"\"\n",
+    "            if t in m.t_con:\n",
+    "                return pyo.Constraint.Skip\n",
+    "            else:\n",
+    "                neighbour_t = max(tc for tc in control_time if tc < t)\n",
+    "                return m.T[t] == m.T[neighbour_t]\n",
+    "\n",
+    "        def cal_kp1(m, t):\n",
+    "            \"\"\"\n",
+    "            Create the perturbation parameter sets\n",
+    "            m: model\n",
+    "            t: time\n",
+    "            \"\"\"\n",
+    "            # LHS: 1/h\n",
+    "            # RHS: 1/h*(kJ/mol *1000J/kJ / (J/mol/K) / K)\n",
+    "            return m.kp1[t] == m.A1 * pyo.exp(-m.E1 * 1000 / (m.R * m.T[t]))\n",
+    "\n",
+    "        def cal_kp2(m, t):\n",
+    "            \"\"\"\n",
+    "            Create the perturbation parameter sets\n",
```

Comment:
This and the function before it appear to do the same thing. What is the difference? Can you clarify the difference either in the name of the function or the docstrings?